elf_reader_test.go

```diff
@@ -474,6 +486,41 @@ func TestLibBPFCompat(t *testing.T) {
 	})
 }
 
+func loadTargetProgram(tb testing.TB, name string, opts CollectionOptions) (*Program, *Collection) {
+	file := "test_pkt_access.o"
+	program := "test_pkt_access"
+	switch name {
```

Comment:
It seems like most freplace selftests use test_pkt_access from test_pkt_access.o, which I suspect will continue to be the case unless they need special functionality.
Also, most of the other tests fail in a way that causes the test to be skipped. I've listed them here because I didn't see a reason not to, but hopefully we won't need to update this for every kernel release.
It's possible, though.
Off the top of my head, I'm not sure of a good way around that (other than strings in the verifier output, there's no way to determine that a BTF mismatch for an extension program is the reason for the EINVAL failure), but I'll give it some thought.